### PR TITLE
Fix pkg.latest on FreeBSD by stripping 'with_origin' dict.

### LINKED ISSUE #55630: state pkg.installed not working when version is specified
I have the following state file.
```
mariadb_pkg:
  pkg.installed:
    - name: mariadb103-server
    - version: {{ salt['pillar.get']('mariadb:version', default_db_version) }}
```
It worked fine for two year until I recently upgraded salt to 2019.2.2

When I run it in debug mode, I got the following message:
```
16:02:31,455 [DEBUG   ] Current version ({'origin': 'databases/mariadb103-server', 'version': ['10.3.20']}) did not match desired version specification (10.3.20), adding to installation targets
```
It seems that salt was trying to compare a `dict` with `str` and decided that it didn't match.

### Versions Report

```
Salt Version:
           Salt: 2019.2.2

Dependency Versions:
           cffi: 1.13.2
       cherrypy: Not Installed
       dateutil: 2.8.0
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.10.1
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.6.2
   mysql-python: 1.4.2.post1
      pycparser: 2.19
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.6.9 (default, Nov 12 2019, 23:28:35)
   python-gnupg: Not Installed
         PyYAML: 5.1
          PyZMQ: 18.1.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.3.1

System Versions:
           dist:
         locale: UTF-8
        machine: amd64
        release: 12.1-RELEASE
         system: FreeBSD
        version: Not Installed
```

### PR DESCRIPTION
**Pull request by krionbsd**
Related to https://github.com/saltstack/salt/pull/54074 and
https://github.com/saltstack/salt/issues/55630

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/55630

### CODE REVIEW

File: salt/states/pkg.py [hunk 1]
```diff
@@ -2975,23 +2965,22 @@ def _uninstall(
...
     if not_installed:
         comments.append(
             "The following packages were not installed: "
-            "{0}".format(", ".join(not_installed))
+            "{}".format(", ".join(not_installed))
         )
         comments.append(
-            "The following packages were {0}d: "
-            "{1}.".format(action, ", ".join(targets))
+            "The following packages were {}d: " "{}.".format(action, ", ".join(targets))
```
Comment (s0undt3ch) [inline salt/states/pkg.py:2980]:
```suggestion
            "The following packages were {}d: {}.".format(action, ", ".join(targets))
```

File: salt/states/pkg.py [hunk 2]
```diff
@@ -3181,7 +3170,7 @@ def purged(name, version=None, pkgs=None, normalize=True, ignore_epoch=None, **k
             ret["changes"] = {}
             ret[
                 "comment"
-            ] = "An error was encountered while purging " "package(s): {0}".format(exc)
+            ] = "An error was encountered while purging " "package(s): {}".format(exc)
```
Comment (s0undt3ch) [inline salt/states/pkg.py:3173]:
```suggestion
            ] = "An error was encountered while purging package(s): {}".format(exc)
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

**Review by Ch3LL — CHANGES_REQUESTED** [review]
tests?

**Comment by sagetherage** [comment]
@waynew maybe we can attempt to write tests for this one at the Test Clinic next week? this one is a good PR example like many of the [Ports to Master](https://github.com/saltstack/salt/pull/59461) PRs that need to be merged, but also need tests written - we should do a write tests code jam.

**Comment by krionbsd** [comment]
> tests?

I think we don't really need a test case for this PR, because what we do for FreeBSD is just strip `with_origin` from the dict, and the `test_fulfills_version_string` test in tests/pytests/unit/states/test_pkg.py covers this scenario for FreeBSD as well

**Review by s0undt3ch — DISMISSED** [review]

**Comment by Ch3LL** [comment]
This is fixing a bug and our previous tests did not catch this, we need to add a tests that fails without this change, and passes with the change.

**Comment by Ch3LL** [comment]
bump on my comment above ^

**Comment by sagetherage** [comment]
@krionbsd when you are back online let's address the testcases here, pretty please

**Pull request closed without merging**